Sources/StreamChat/Workers/ThreadListUpdater.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+
+/// Makes a thread list query call to the backend and updates the local storage with the results.
+class ThreadListUpdater: Worker {
```

Comment:
I would prefer if we use repositories for the new features. We anyway want to migrate away from the workers.

Reply:
With our current architecture, our Updaters are already our Repositories, IMO introducing the same thing but with a different name just causes incosistencies (like we have the moment). If you check the `MessageRepository`, it is pretty much the same thing as the Updaters, but with a different name. The state layer is also using the Updaters for now. So I would only change this when we start working on v5. For now I would favor consistency

Reply:
We also just add the `ReactionListController` which was also created with `ReactionListUpdater`, so it does not make sense for this one to be different IMO

Reply:
I would prefer to use Repositories tbh, also for the reactions, which is probably something I missed in the review (probably because of the urgency of the feature). Now we're in a state where we migrated some stuff, and then we keep going back to the old names again.